### PR TITLE
[WIP] Add tutorials on UDEs with neural network rates

### PR DESCRIPTION
**Pull request by TorkelE**
Merging will likely require https://github.com/SciML/Catalyst.jl/pull/1488 to be finished first (which in turn is waiting for the next OrdinaryDiffEq release).

### CODE REVIEW

File: docs/src/inverse_problems/udes.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+
+# Create a UDE using Lux and ModelingToolkitNeuralNets.
+using ModelingToolkitNeuralNets, Lux
+nn_arch = Lux.Chain(
+    Lux.Dense(1 => 3, Lux.softplus, use_bias = false),
+    Lux.Dense(3 => 3, Lux.softplus, use_bias = false),
+    Lux.Dense(3 => 1, Lux.softplus, use_bias = false),
+)
+@SymbolicNeuralNetwork U, θ = nn_arch
+A(z) = U(z, θ)[1]
```
Comment (sebapersson) [inline docs/src/inverse_problems/udes.md:61]: Maybe use U([z, ]) syntax, makes it easier to understand how to have more than a single input to the neural network

File: docs/src/inverse_problems/udes.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+```
+We now generate a `SymbolicNeuralNetwork` using ModelingToolkitNeuralNets. This embeds the
+Lux model in a symbolic format that is compatible with Catalyst (and ModelingToolkit).
+```@example ude_rate_based
+using ModelingToolkitNeuralNets
+@SymbolicNeuralNetwork U, θ = nn_arch
+```
+Here, `U` represents the neural network structure, and `θ` represents its parameters. Additional options for using the `@SymbolicNeuralNetwork` macro are described [here](https://sciml.github.io/ModelingToolkitNeuralNets.jl/stable/api/#ModelingToolkitNeuralNets.@SymbolicNeuralNetwork) and [here](https://sciml.github.io/ModelingToolkitNeuralNets.jl/stable/symbolic_ude_tutorial/). Next, using  the following syntax, we can [interpolate](@ref dsl_advanced_options_symbolics_and_DSL_interpolation) this function approximator as  a Catalyst model rate.
+```@example ude_rate_based
+A(z) = U(z, θ)[1]
```
Comment (sebapersson) [inline docs/src/inverse_problems/udes.md:176]: See comment above

File: docs/src/inverse_problems/udes.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+nothing # hide
+```
+PEtab supports many options for customising the optimisation problem. Here, we choose the
+simplest settings. These options are described further in the [Catalyst-specific PEtab
+tutorial](@ref petab_parameter_fitting), and in
+[PEtab's documentation](https://sebapersson.github.io/PEtab.jl/stable/).
+
+The UDE model can now be fitted to the data. Here, we use the Adam implementation from the
+[Optimisers.jl](https://github.com/FluxML/Optimisers.jl) package, running 5 independent starts 
+with 10,000 iterations each.
```
Comment (sebapersson) [inline docs/src/inverse_problems/udes.md]:
```suggestion
with 10,000 iterations/epochs each.
```

File: docs/src/inverse_problems/udes.md [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+an approach similar to those used [here](@ref optimization_parameter_fitting) and
+[here](@ref behaviour_optimisation) to declare a loss function comparing the behaviour of
+this UDE with available data. Next, the parameters can be fitted by minimising that loss
+function. An example of how this is done for an ODE-declared UDE can be found
+[here](https://sciml.github.io/ModelingToolkitNeuralNets.jl/stable/symbolic_ude_tutorial/).
+
+### [Learning parameters or observables using neural networks](@id udes_parameters_n_observables)
+Throughout this tutorial, we have shown how neural networks can be incorporated into
+Catalyst models to learn unknown functions of system variables within the main model
+system. PEtab, however, also supports two additional ways to incorporate neural networks
```
Comment (sebapersson) [inline docs/src/inverse_problems/udes.md]:
```suggestion
system. PEtab.jl, however, also supports two additional ways to incorporate neural networks
```

**Review by sebapersson — COMMENTED** [review]
Thanks for adding, I think it looks good! Left a few minor comments

**Comment by TorkelE** [comment]
This one needs a bit of tuning still
